client/asset/btc/btc.go

```diff
@@ -429,78 +572,100 @@ type findRedemptionResult struct {
+	case dex.Regtest:
+		return &chaincfg.RegressionNetParams, nil
+	}
+	return nil, fmt.Errorf("unknown network ID %v", net)
+}
+
 // NewWallet is the exported constructor by which the DEX will import the
 // exchange wallet. The wallet will shut down when the provided context is
 // canceled. The configPath can be an empty string, in which case the standard
 // system location of the bitcoind config file is assumed.
-func NewWallet(cfg *asset.WalletConfig, logger dex.Logger, network dex.Network) (asset.Wallet, error) {
-	var params *chaincfg.Params
-	switch network {
-	case dex.Mainnet:
-		params = &chaincfg.MainNetParams
-	case dex.Testnet:
-		params = &chaincfg.TestNet3Params
-	case dex.Regtest:
-		params = &chaincfg.RegressionNetParams
-	default:
-		return nil, fmt.Errorf("unknown network ID %v", network)
+func NewWallet(cfg *asset.WalletConfig, logger dex.Logger, net dex.Network) (asset.Wallet, error) {
+	params, err := parseChainParams(net)
+	if err != nil {
+		return nil, err
 	}
+
 	cloneCFG := &BTCCloneCFG{
 		WalletCFG:           cfg,
 		MinNetworkVersion:   minNetworkVersion,
 		WalletInfo:          WalletInfo,
 		Symbol:              "btc",
 		Logger:              logger,
-		Network:             network,
+		Network:             net,
 		ChainParams:         params,
 		Ports:               dexbtc.RPCPorts,
 		DefaultFallbackFee:  defaultFee,
 		DefaultFeeRateLimit: defaultFeeRateLimit,
 		Segwit:              true,
 	}
 
-	return BTCCloneWallet(cloneCFG)
+	switch cfg.Type {
+	case WalletTypeSPV:
+		if atomic.LoadUint32(&initialized) == 0 {
+			return nil, fmt.Errorf("must call asset.Initialize if you want to create an SPV wallet")
+		}
+		return openSPVWallet(cloneCFG)
+	case WalletTypeRPC, WalletTypeLegacy:
+		return BTCCloneWallet(cloneCFG)
+	default:
+		return nil, fmt.Errorf("unknown wallet type %q", cfg.Type)
+	}
 }
 
 // BTCCloneWallet creates a wallet backend for a set of network parameters and
 // default network ports. A BTC clone can use this method, possibly in
 // conjunction with ReadCloneParams, to create a ExchangeWallet for other assets
 // with minimal coding.
 func BTCCloneWallet(cfg *BTCCloneCFG) (*ExchangeWallet, error) {
-	// Read the configuration parameters
-	btcCfg, err := dexbtc.LoadConfigFromSettings(cfg.WalletCFG.Settings,
-		cfg.Symbol, cfg.Network, cfg.Ports)
+	clientCfg, client, err := ParseRPCWalletConfig(cfg.WalletCFG.Settings, cfg.Symbol, cfg.Network, cfg.Ports)
 	if err != nil {
 		return nil, err
 	}
 
-	endpoint := btcCfg.RPCBind + "/wallet/" + cfg.WalletCFG.Settings["walletname"]
-	cfg.Logger.Infof("Setting up new %s wallet at %s.", cfg.Symbol, endpoint)
+	btc, err := newRPCWallet(client, cfg, &clientCfg.WalletConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error creating %s ExchangeWallet: %v", cfg.Symbol,
+			err)
+	}
+
+	return btc, nil
+}
 
-	client, err := rpcclient.New(&rpcclient.ConnConfig{
+func newRPCWalletConnection(cfg *RPCWalletConfig) (*rpcclient.Client, error) {
```

Comment:
This function looks simple and is only called from `ParseRPCWalletConfig`, can probably inline it there instead.